src/expressions/nodes/ParentNode.cpp

```diff
@@ -33,6 +33,16 @@ const std::vector<Node*>& ParentNode::getOperands() const
     return operands_;
 }
 
+const std::vector<const Node*> ParentNode::getConstOperands() const
+{
+    std::vector<const Node*> constOperands;
+    for (const auto* operand: operands_)
+    {
+        constOperands.push_back(operand);
+    }
+    return constOperands;
+}
+
```

Comment:
1. **Naming** You can have 2 member functions with the same name but different constness. The compiler will get the correct one without any problem
2. **Style** You can construct a `std::vector` from two iterators

```suggestion
const std::vector<const Node*> ParentNode::getOperands() const
{
    std::vector<const Node*> constOperands(operands_.begin(), operands_.end());
    return constOperands;
}

```

Reply:
3. **Style** For a function `f`, `ReturnType f()` and `const ReturnType f()` are equivalent

Reply:
Functions that differ only in their return type cannot be overloaded                                                                                                                                                                  
ParentNode.h:74:31: note: previous declaration is here [ovl_diff_return_type]